content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -938,6 +938,16 @@ creates Pods with the finalizer `batch.kubernetes.io/job-tracking`. The
 controller removes the finalizer only after the Pod has been accounted for in
 the Job status, allowing the Pod to be removed by other controllers or users.
 
+#### Troubleshooting
+
+- **Pods stuck in Terminating state and Jobs don't complete**
+
+  Check for validating or mutating webhooks for Pod updates. Some faulty
```

Comment:
Calling such webhooks "faulty" is a bit strong here. Sometimes validating or mutating webhooks are configured (by users, either explicitly or implicitly) to match on UPDATE requests for entirely valid and useful purposes. In some cases, this can have undesirable consequences which is the case with just about every thing else.

Reply:
I changed it to "misconfigured or faulty"

Reply:
It still seems like you're suggesting a webhook which is configured to react to UPDATE requests _must_ be one of either "misconfigured" or "faulty" and that's not true. I get what you're trying to convey, but it could be stated to better reflect the variety of webhooks and their driving use cases. Here's my suggestion:

> "Webhooks which are configured to match on UPDATE requests may in inadvertently deny an update for a Pod that is already running or finished. This causes kube-apiserver to reject the update to remove the Pod finalizers, preventing the Job controller from making process. Inspect and disable or modify the webhook or its managing controller's configuration to allow these updates."

Reply:
This might not be the case of kyverno, but a webhook that tries to add an initContainer to a running or finished Pod (https://github.com/kubernetes/kubernetes/issues/121605) is a faulty webhook, because in under no circumstances that works.

If you configure a webhook to reject Pod updates for existing Pods, that's a misconfiguration in the user side.

Reply:
The webhook, as an API resource, is not the issue here. It's what the service does with the AdmissionReview that it receives as a result of the webhook's configuration. A webhook can only ask for UPDATE ops; it has no logic to do anything with them. Having a webhook which matches on UPDATE to Pods is neither "faulty" or represents a "misconfiguration." Having logic which attempts to add an initContainer, per your example, based upon the call-back it receives, is "faulty" or does represent a "misconfiguration."

Reply:
Here I am referring to webhook as the combination the Webhook object and the service that responds to it. I don't think most users make the distinction between the two.

@sftim do you have any guidelines?

Reply:
I think some clarification may be order lest this paragraph inspire users to go inspect all their webhooks and, upon finding any that match on UPDATE ops, proceed to <project/product> and proclaim it's a bug and shouldn't be there. The guidance is helpful for sure, but more specifics are needed to avoid making wrong assumptions.